### ISSUE TITLE
Missing header causes build failures for c10d extensions

### ISSUE THREAD

**Issue by lw** [pytorch/pytorch#73421]
### 🐛 Describe the bug

(This issue is a standalone copy of [my comment](https://github.com/pytorch/pytorch/issues/72267#issuecomment-1050793837))

PR https://github.com/pytorch/pytorch/pull/73144 introduced a regression which makes it impossible to build c10d extensions, because it adds a `c10d/debug.h` header but it doesn't install it (but which is recursively included from `c10d/ProcessGroup.hpp`). [This](https://app.circleci.com/pipelines/github/facebookresearch/fairring/343/workflows/24b8e6f8-b5f2-41d2-b428-b054e2c100c8/jobs/4416) is the failure I'm getting in my CI. It is imperative to fix this before the v1.11 stable release. I'm surprised this passed CI: I thought we had a test that builds a simple c10d extension?

EDIT: It's possible that it's just because the header's extension is `.h` instead of `.hpp` like other files in that directory (legacy reasons...). Perhaps all that's needed is to update this filter?
https://github.com/pytorch/pytorch/blob/285272f399561cb0a3b5b6a48225d147944fcdf5/caffe2/CMakeLists.txt#L1111-L1113

### Versions

N/A

cc @ezyang @gchanan @zou3519 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

**Comment by lw** [comment]
Copying @cbalioglu's comment:

@lw the original PR is https://github.com/pytorch/pytorch/pull/73144. As you can see both internal and external CIs were green. Do you know where that test is defined?

> EDIT: It's possible that it's just because the header's extension is .h instead of .hpp like other files in that directory (legacy reasons...). Perhaps all that's needed is to update this filter?

Yes, that is most likely the reason. I can obviously fix it, but we have to figure out why our CI did not catch this in the initial PR.

**Comment by lw** [comment]
The test I had in mind was the one that uses this file: https://github.com/pytorch/pytorch/blob/master/test/cpp_extensions/cpp_c10d_extension.hpp

However it looks like that file isn't currently referenced from anywhere.

I did some code archeology and it appears to me that the test that was building that file was removed in #63454. CC @pritamdamania87 to check whether that was intentional or just a mishap during a refactoring.
